### PR TITLE
Supports solo-1.x

### PR DESCRIPTION
**Pull request by tcr3dr**
This incorporates the "pysim" code from APM to run older versions of ArduCopter, i.e. those that need an external Python script for operation.

It uses only the pysim command line interface to skirt any potential licensing issues, though it also dual-licenses the code under GPL to reflect this change. (Permitted via the pre-existing MIT license.)

This also incorporates some stub code for loading params on bootup, which will be extended to support all platforms later. This support requires DroneKit as a dependency now, which makes sense.

### CODE REVIEW

File: dronekit_sitl/__init__.py [hunk 1]
```diff
@@ -183,6 +229,22 @@ def cleanup():
...
+        # Run dronekit
+        if need_sim:
+            time.sleep(0.5)
+            vehicle = dronekit.connect('tcp:127.0.0.1:5760')
+            for line in open(os.path.join(os.path.dirname(__file__), 'defaults.parm')):
+                if re.match(r'^\s*#', line):
+                    continue
+                try:
+                    pname, pvalue = line.split()
+                    vehicle.parameters.set(pname, float(pvalue), retries=0)
```
Comment (mrpollo) [inline dronekit_sitl/__init__.py]: :+1:

**Comment by tcr3dr** [comment]
Don't merge, found an error on Windows...

**Comment by hamishwillee** [comment]
@tcr3dr `default.parm` file (and hence parameters) are build/vehicle specific - fetched from autotest at build time?

**Comment by mrpollo** [comment]
no, we have defaults for solo, we need to get defaults for each vehicle and version, can you help us put together a matrix of params needed?

**Comment by hamishwillee** [comment]
The right parameters will always be the associated **.parms** file for the vehicle in  **/ardupilot/Tools/autotest** at build time:
copter_params.parm[](https://github.com/diydrones/ardupilot/blob/master/Tools/autotest/copter_params.parm)
[ArduPlane.parm](https://github.com/diydrones/ardupilot/blob/master/Tools/autotest/ArduPlane.parm)
[Rover.parm](https://github.com/diydrones/ardupilot/blob/master/Tools/autotest/Rover.parm)

It would be best if we could package these at build time for each vehicle download. 

Does that answer your question?

**Comment by mrpollo** [comment]
not really, we know where to get them, what we need is a list of supported versions we should have, I guess for each ardu-project we build we should keep the param files, 

@tcr3dr wheres the dronekit-sitl-builder repo?

**Comment by hamishwillee** [comment]
Yes we need to keep the param files for each build - perhaps if this can be dumped in the right place and renamed default.parm everything will work out of the box.

As to what versions we need to build, I think current and last stable releases for rover, copter, plane and solo. I don't think we need any dev builds.

**Comment by tcr3dr** [comment]
We're going to package them inside of the .tar.gz builds that SITL has. Those all need to be rebuilt.

**Pull request merged**